### PR TITLE
Use string catalogues and fix non-localized strings

### PR DESCRIPTION
**Pull request by vishnuravi**
# Use string catalogues and fix non-localized strings

## :gear: Release Notes 
Migrates to use of string catalogues for localization and fixes a string that is not being drawn from the localization files.

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [X] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vishnuravi** [comment]
@Supereg any idea why we are getting the `discouraged_optional_collection` error here? Those files were not touched and seemed to have passed before?

**Comment by Supereg** [comment]
> @Supereg any idea why we are getting the `discouraged_optional_collection` error here? Those files were not touched and seemed to have passed before?

Not sure why they pop up now. But feel free to add some `//swiftlint:disable:this discouraged_optional_collection` to the lines.

**Review by Supereg — APPROVED** [review]
Thank you 🚀

**Comment by vishnuravi** [comment]
> > @Supereg any idea why we are getting the `discouraged_optional_collection` error here? Those files were not touched and seemed to have passed before?
> 
> Not sure why they pop up now. But feel free to add some `//swiftlint:disable:this discouraged_optional_collection` to the lines.

Thanks. I added this and then I got a line length - sometimes you just can't win!!

**Comment by Supereg** [comment]
> > > @Supereg any idea why we are getting the `discouraged_optional_collection` error here? Those files were not touched and seemed to have passed before?
> > 
> > 
> > Not sure why they pop up now. But feel free to add some `//swiftlint:disable:this discouraged_optional_collection` to the lines.
> 
> Thanks. I added this and then I got a line length - sometimes you just can't win!!

There is also a `swiftlint:disable:previous ...` which might be helpful here 😅

**Comment by vishnuravi** [comment]
> > > > @Supereg any idea why we are getting the `discouraged_optional_collection` error here? Those files were not touched and seemed to have passed before?
> > > 
> > > 
> > > Not sure why they pop up now. But feel free to add some `//swiftlint:disable:this discouraged_optional_collection` to the lines.
> > 
> > 
> > Thanks. I added this and then I got a line length - sometimes you just can't win!!
> 
> There is also a `swiftlint:disable:previous ...` which might be helpful here 😅

Fixed the SwiftLint error and got a REUSE error. Tonight is just not my night Andreas. 😅

**Comment by PSchmiedmayer** [comment]
@vishnuravi @Supereg Thank you for this fix!

Can we use this PR to directly resolve #22 which would have also been helpful to be done right in #32 to ensure that we don't run into the same issues again?
String catalogues should flag these issues and should automate these processes in the future 👍

**Review by PSchmiedmayer — CHANGES_REQUESTED** [review]
@vishnuravi @Supereg Thank you for this fix!

Can we use this PR to directly resolve #22 which would have also been helpful to be done right in #32 to ensure that we don't run into the same issues again?
String catalogues should flag these issues and should automate these processes in the future 👍

File: Sources/SpeziAccount/Resources/Localizable.xcstrings [hunk 1]
```diff
@@ -0,0 +1,2090 @@
+{
+  "sourceLanguage" : "en",
+  "strings" : {
+    "" : {
+
+    },
+    ", " : {
+
+    },
```
Comment (Supereg) [inline Sources/SpeziAccount/Resources/Localizable.xcstrings:9]: Can we check where this two translations are coming from?
Seems like there are some `Text` views that do not use the [Text(verbatim:)](https://developer.apple.com/documentation/swiftui/text/init(verbatim:)) initializer.

File: Sources/SpeziAccount/Resources/Localizable.xcstrings [hunk 2]
```diff
@@ -0,0 +1,2090 @@
...
+          }
+        }
+      }
+    },
+    "Continue" : {
+
+    },
+    "DEFAULT" : {
+
+    },
```
Comment (Supereg) [inline Sources/SpeziAccount/Resources/Localizable.xcstrings:616]: Seems like there are also missing translations or misused Text initializers?

File: Sources/SpeziAccount/Resources/Localizable.xcstrings [hunk 3]
```diff
@@ -0,0 +1,2090 @@
...
+    "Hello" : {
+
+    },
+    "Hello World" : {
+
+    },
+    "License Information" : {
+
+    },
```
Comment (Supereg) [inline Sources/SpeziAccount/Resources/Localizable.xcstrings:927]: If these are used within the Previews, we might just rewrite the previews to use the `Text(verbatim:)` initializer?

File: Sources/SpeziAccount/Resources/Localizable.xcstrings [hunk 4]
```diff
@@ -0,0 +1,2090 @@
...
+    "Mock AccountService" : {
+
+    },
+    "Mock SignIn with Apple" : {
+
+    },
+    "Mock Simple AccountService" : {
+
+    },
```
Comment (Supereg) [inline Sources/SpeziAccount/Resources/Localizable.xcstrings:982]: Yep, seems like I forgot to use the `Text(verbatim:)` initializer all the Mock implementations.

File: Sources/SpeziAccount/Resources/Localizable.xcstrings [hunk 5]
```diff
@@ -0,0 +1,2090 @@
...
+    "Package Dependencies" : {
+
+    },
+    "Password Text" : {
+
+    },
+    "Preview Service" : {
+
+    },
```
Comment (Supereg) [inline Sources/SpeziAccount/Resources/Localizable.xcstrings:1129]: here as well

File: Sources/SpeziAccount/Resources/Localizable.xcstrings [hunk 6]
```diff
@@ -0,0 +1,2090 @@
...
+    "Some Hint" : {
+
+    },
+    "The internal account identifier is meant to be generated!" : {
+
+    },
+    "The internal account identifier is not meant to be user facing!" : {
+
+    },
```
Comment (Supereg) [inline Sources/SpeziAccount/Resources/Localizable.xcstrings:1233]: Seems like either missing translation or again missing use of the `Text(verbatim:)` initializer

File: Sources/SpeziAccount/Resources/Localizable.xcstrings [hunk 7]
```diff
@@ -0,0 +1,2090 @@
...
+          "stringUnit" : {
+            "state" : "translated",
+            "value" : "Agregar %@"
+          }
+        }
+      }
+    },
+    "World" : {
+
+    },
```
Comment (Supereg) [inline Sources/SpeziAccount/Resources/Localizable.xcstrings:2064]: Probably a string coming from the previews?

**Review by Supereg — APPROVED** [review]
Thanks for the great work 🚀 

As we migrated to String catalogues, we have some strings being picked up from the mock or preview implementations. I annotated every localization key I could find.
Would be great if you could review the usages of those keys and check if it makes sense to use the `Text(verbatim:)` initializer instead, or, otherwise, add a translation for those strings.

Otherwise, the PR looks great, thanks for the effort 🎉

**Review by PSchmiedmayer — APPROVED** [review]
Thank you @vishnuravi! Happy to get the PR merged once @Supereg is happy and his comments are resolved 🚀

File: Package.swift
Comment (PSchmiedmayer) [inline Package.swift:1]: Would be great if we can also just include the MD link check in there as well, just so we get it into the different repos as we make different smaller or bigger PRs 👍

**Comment by vishnuravi** [comment]
> Thank you @vishnuravi! Happy to get the PR merged once @Supereg is happy and his comments are resolved 🚀

@Supereg and I are having a side chat on Slack, we are not sure it is possible to designate strings like "Mock Simple AccountService" below as non-localized:

```
let configuration = AccountServiceConfiguration(name: "Mock Simple AccountService", supportedKeys: .arbitrary)
```

Any thoughts there?

**Comment by PSchmiedmayer** [comment]
@vishnuravi Following up on our discussion today: I addressed most of the issues with the localization files and only needed to translate a few very limited elements and added comments for the rest.

Looking forward to getting the PR finished and feel free to merge it once all the conversations are resolved 👍

**Comment by Supereg** [comment]
fyi, I'm currently rebasing this PR and taking care of it getting it merged if that is alright for you @vishnuravi 👍

**Comment by vishnuravi** [comment]
> fyi, I'm currently rebasing this PR and taking care of it getting it merged if that is alright for you @vishnuravi 👍 

That sounds great, thank you @Supereg!

**Review by PSchmiedmayer — APPROVED** [review]
Thank you for finishing the PR @Supereg and thanks for all the work here @vishnuravi!

**Pull request merged**